lib/libyui_client/timer.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  module Timer
+    module_function
+
+    def wait(timeout, interval)
+      return yield if timeout.zero?
+
+      end_time = Time.now.to_f + timeout
```

Comment:
Why not to use `to_i` in both cases as we won't need big precision. Using `to_f` in both cases would be fine too.